Neos.Flow/Classes/Http/BaseRequest.php

```diff
@@ -0,0 +1,389 @@
+<?php
+namespace Neos\Flow\Http;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use Neos\Utility\MediaTypes;
+
+/**
+ *
+ */
+class BaseRequest extends AbstractMessage implements RequestInterface
+{
+    /**
+     * @var Uri
+     */
+    protected $uri;
+
+    /**
+     * @var Uri
+     */
+    protected $baseUri;
+
+    /**
+     * @var string
+     */
+    protected $method = 'GET';
+
+    /**
+     * URI for the "input" stream wrapper which can be modified for testing purposes
+     *
+     * @var string
+     */
+    protected $inputStreamUri = null;
+
+    /**
+     * @var string
+     */
+    protected $requestTarget = null;
+
+    /**
+     * BaseRequest constructor.
+     *
+     * @param UriInterface $uri
+     * @param string $method
+     */
+    public function __construct(UriInterface $uri, $method = 'GET')
```

Comment:
Many implementations, including our current HTTP Request default to GET, so I felt it sensible to apply the same here.